core/frontend/src/render/webgl/BranchUniforms.ts

```diff
@@ -103,6 +104,19 @@ export class BranchUniforms {
   public pushBranch(branch: Branch): void {
     desync(this);
     this._stack.pushBranch(branch);
+
+    const vp = IModelApp.viewManager.selectedView;
+    if (vp) {
+
+      /** If the branch has disableClipStyle set, we set these alphas to 0, therefore disabling the clip style.
+      * Otherwise, we leave it to the viewport's clip style.
+      */
+      const style = vp.view.displayStyle.settings.clipStyle;
```

Comment:
It looks to me like if I have a graphic with the following structure:
```
Branch: disableClipStyle = true
  Branch: disableClipStyle = false (or undefined)
    Some visible graphics
```

Then when you push the second branch you will turn the clip style back on, where we want the disableClipStyle flag to be inherited from the top of the stack when pushing, and restored when popping.

Can you write a test that proves me wrong?

Reply:
You were right, some logic has been added to `BranchUniforms.pushBranch` to ensure `disableClipStyle` is inherited from the top of the stack, and a test for this has been added to `BranchUniforms.test.ts`.

Reply:
Your test doesn't appear to validate the "resotre when popping" behavior and  your implementation here doesn't appear to handle it.

If I have this scene graph, where only branch A1 disables the clip style:
```
Branch A
  Branch A1: disableClipStyle = true
    Branch A1a
  Branch A2
Branch B
  Branch B1
```

and the view has a clip style, then the clip style should be applied to branches A2, B, and B1; and not applied to branches A1 and A1a.

AFAICT, the way you've implemented it, as soon as any branch disables the clip style, it remains disabled for everything subsequently drawn, even after that branch is popped. Did I overlook something?

Reply:
You're right, some changes are in progress. A quick question, assume we had this graph:
```
Branch A
  Branch A1: disableClipStyle = true
    Branch A1a
      Branch1aa
    Branch1b
  Branch A2
Branch B
  Branch B1
```

In this scenario, am I correct in saying we want both Branch1aa and Branch1b to both have a disabled clip style?

Reply:
Yep. It's a stack. State changes on push, resets on pop.

Reply:
In `BranchUniforms.ts`, logic has been added to `pushBranch` to ensure `disableClipstyle` is inherited even if there are multiple branches in a row with it undefined like the scenario above,  logic has been added to `pop` to restore the clipstyle when a branch is popped, and two new functions `setClipStyle` and `resetClipStyle` were created for clarity within `pushBranch`. `testBranches` in `BranchUniforms.test.ts` now also optionally tests for the clipstyle being restored after a branch is popped.